### PR TITLE
Improve messages emitted when Reload or Reexec is requested

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: src/core/dbus-manager.c [hunk 1]
```diff
@@ -1527,8 +1527,10 @@ static void log_caller(sd_bus_message *message, Manager *manager, const char *me
         (void) sd_bus_creds_get_comm(creds, &comm);
         caller = manager_get_unit_by_pid(manager, pid);
 
-        log_info("%s requested from client PID " PID_FMT " ('%s') (from unit '%s')...",
-                 method, pid, strna(comm), strna(caller ? caller->id : NULL));
+        log_info("%s requested from client PID " PID_FMT "%s%s%s%s%s%s...",
+                 method, pid,
+                 comm ? " ('" : "", strempty(comm), comm ? "')" : "",
+                 caller ? " (unit " : "", caller ? caller -> id : NULL, caller ? ")" : "");
```
Comment (bluca) [inline src/core/dbus-manager.c]: `caller -> id` looks a bit strange with the spaces?

Comment (mrc0mmand) [reply src/core/dbus-manager.c]: Maybe `strjoina()` would make this slightly easier on the eyes?

File: src/core/service.c [hunk 2]
```diff
@@ -2763,7 +2763,13 @@ static int service_serialize_exec_command(Unit *u, FILE *f, const ExecCommand *c
                 return log_oom();
 
         key = strjoina(type, "-command");
-        (void) serialize_item_format(f, key, "%s %u %s %s", service_exec_command_to_string(id), idx, p, args);
+        (void) serialize_item_format(
+                        f, key,
+                        "%s %s%u %s %s",
+                        service_exec_command_to_string(id),
+                        command->command_next ? "" : "+",
```
Comment (poettering) [inline src/core/service.c:2773]: this is mean. but maybe ok. but please add a comment explaining this hack here to the sources

File: src/core/service.c [hunk 3]
```diff
@@ -2921,6 +2927,7 @@ int service_deserialize_exec_command(
                         r = safe_atou(arg, &idx);
                         if (r < 0)
                                 return r;
+                        last = arg[0] == '+';
```
Comment (poettering) [inline src/core/service.c:2938]: similar, add a comment here

File: src/libsystemd/sd-bus/sd-bus.c [hunk 4]
```diff
@@ -48,7 +48,9 @@
 #define log_debug_bus_message(m)                                         \
         do {                                                             \
                 sd_bus_message *_mm = (m);                               \
-                log_debug("Got message type=%s sender=%s destination=%s path=%s interface=%s member=%s cookie=%" PRIu64 " reply_cookie=%" PRIu64 " signature=%s error-name=%s error-message=%s", \
```
Comment (poettering) [inline src/libsystemd/sd-bus/sd-bus.c:51]: the commit msg is a bit misleading. this is not about indentation, but about line breaking

File: src/core/dbus-manager.c (hunk 1 above)
Comment (poettering) [reply src/core/dbus-manager.c]: yeah, the spaces should just go.

Comment (keszybz) [reply src/core/dbus-manager.c]: > Maybe `strjoina()` would make this slightly easier on the eyes?

`strjoina()` cannot be used in an argument list… So we'd need to use a temporary variable to assign the result. I don't think that'd be better.

**Comment by keszybz** [comment]
Thank you for the review. Updated with the suggested changes.

**Review by bluca — APPROVED** [review]

File: src/core/dbus-manager.c (hunk 1 above)
Comment (mrc0mmand) [reply src/core/dbus-manager.c]: > > Maybe `strjoina()` would make this slightly easier on the eyes?
> 
> `strjoina()` cannot be used in an argument list… So we'd need to use a temporary variable to assign the result. I don't think that'd be better.

Ah, I see. In that case it really doesn't make sense.

**Comment by bluca** [comment]
@mrc0mmand have you seen this fail before in CentOS CI (Arch Linux + sanitizers)?

`543/1434 systemd:test / test-async      TIMEOUT         360.33s`

**Pull request merged**